201-managed-application-with-customprovider/artifacts/ManagedAppZip/mainTemplate.json

```diff
@@ -0,0 +1,156 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "eastus",
+            "allowedValues": [ "australiaeast", "eastus", "westeurope" ],
+            "metadata": {
+                "description": "Location for the resources."
+            }
+        },
+        "funcname": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the azure fuction that hosts the code. Must be globally unique"
+              },
+            "defaultValue": ""
+        },
+        "storageName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the storage account that hosts the function. Must be globally unique"
+              },
+            "defaultValue": ""
+        },
+        "zipFileBlobUri": {
+          "type": "string",
+          "metadata": {
+            "description": "The Uri for the uploaded function zip file"
+          },
+          "defaultValue": "https://github.com/Azure/azure-quickstart-templates/blob/master/201-managed-application-with-customprovider/artifacts/functionzip/functionpackage.zip?raw=true"
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts